lib/scala/kg/src/main/scala/data/kg/KgtkData.scala

```diff
@@ -0,0 +1,38 @@
+package data.kg
+
+import models.kg.{KgEdge, KgNode}
+import scala.collection.mutable.HashMap
+
+class KgtkData(edges: List[KgEdge], nodes: List[KgNode])
+  extends KgData(edges, nodes, List()) {
+
+  def this(data: Tuple3[List[KgEdge], List[KgNode], List[KgNode]]) =
+    this(edges = data._1, nodes = KgtkData.reduceNodes(data._2, data._3))
+
+  def this(resource: KgtkDataResource) =
+    this(data = resource.read())
+}
+
+object KgtkData {
+  def reduceNodes(nodes1: List[KgNode], nodes2: List[KgNode]): List[KgNode] = {
+    val nodesById = HashMap[String, KgNode]()
```

Comment:
Nit: in general you're supposed to avoid mutable data structures in Scala, but this isn't too bad. No need to change it.

Reply:
Got it, would there be another way of doing this that would use immutable data structures or be more Scala-esque?